### PR TITLE
Support multiple publishers

### PR DESCRIPTION
**Pull request by dominiklohmann**
This changes the `publish` and `subscribe` operators to be MPMC rather than SPMC.

Additionally, this brings a major performance boost to the operators. The cost of transfer from `subscribe` to a matching `publish` should be effectively zero now, making inter-pipeline event transfer as cheap as intra-pipeline event transfer.

Fixes https://github.com/tenzir/issues/issues/1877

### CODE REVIEW

File: changelog/next/features/4270--mpmc-pub-sub.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+The `publish` operator's topic must no longer be unique. Instead, any number of
```
Comment (mavam) [inline changelog/next/features/4270--mpmc-pub-sub.md:1]:
```suggestion
The `publish` operator's topic no longer have to be unique. Instead, any number of
```

Comment (mavam) [reply changelog/next/features/4270--mpmc-pub-sub.md:1]: With "must not" you basically say they can't be unique. Not what you want, I think.

File: changelog/next/features/4270--mpmc-pub-sub.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+pipelines may use the `publish` operator with the same topic. This enables
+multi-producer multi-consumer event routing, where streams of events from
```
Comment (mavam) [inline changelog/next/features/4270--mpmc-pub-sub.md:3]:
```suggestion
multi-producer, multi-consumer (MPMC) event routing, where streams of events from
```

**Review by mavam — APPROVED** [review]

File: changelog/next/features/4270--mpmc-pub-sub.md (hunk 1 above)
Comment (dominiklohmann) [reply changelog/next/features/4270--mpmc-pub-sub.md]: Good point, not sure what I was thinking yesterday while writing this. 😅

File: contrib/tenzir-plugins
Comment (tobim) [inline contrib/tenzir-plugins:1]: Please address the review comments in https://github.com/tenzir/tenzir-plugins/pull/250#pullrequestreview-2095845896.

**Review by tobim — APPROVED** [review]

**Pull request merged**